### PR TITLE
TablePanel: Sort numbers correctly 

### PR DESCRIPTION
**Pull request by speakyourcode**
This PR fixes issue #14781

### CODE REVIEW

File: packages/grafana-ui/src/components/Table/utils.ts [hunk 1]
```diff
@@ -58,6 +58,7 @@ export function getColumns(data: DataFrame, availableWidth: number, columnMinWid
       accessor: (row: any, i: number) => {
         return field.values.get(i);
       },
+      sortType: 'basic',
```
Comment (torkelo) [inline packages/grafana-ui/src/components/Table/utils.ts:61]: should this property not depend on field type? (so be different for number or string fields)

Comment (speakyourcode) [reply packages/grafana-ui/src/components/Table/utils.ts:61]: Something like this? I'm not sure what sorting to use for the other types (boolean, trace...)

```ts
   const selectSortType = (type: FieldType): string => {
      switch (type) {
        case FieldType.number: return 'basic';
        case FieldType.time : return 'datetime';
        default: return 'alphanumeric';
      }
    }

    const Cell = getCellComponent(fieldTableOptions.displayMode, field);
    columns.push({
      Cell,
      id: fieldIndex.toString(),
      Header: getFieldDisplayName(field, data),
      accessor: (row: any, i: number) => {
        return field.values.get(i);
      },
      sortType: selectSortType(field.type),
      width: fieldTableOptions.width,
      minWidth: 50,
    });
```

Comment (torkelo) [reply packages/grafana-ui/src/components/Table/utils.ts:61]: yes! think time should also be numeric (as time fields are mostly number ms epochs)

**Comment by torkelo** [comment]
Thanks for contributing!

**Comment by himao** [comment]
The transform "Convert field type" can be a workaround.

**Pull request merged**